inter/block.go

```diff
@@ -15,12 +16,33 @@ type Block struct {
 	SkippedTxs  []uint32      // indexes of skipped txs, starting from first tx of first event, ending with last tx of last event
 	GasUsed     uint64
 	Root        hash.Hash
+	prevRandao  common.Hash
 }
 
 func (b *Block) EstimateSize() int {
 	return (len(b.Events)+len(b.InternalTxs)+len(b.Txs)+1+1)*32 + len(b.SkippedTxs)*4 + 8 + 8
 }
 
+func (b *Block) SetPrevRandao(prevrandao common.Hash) {
+	b.prevRandao = prevrandao
+}
+
+func (b *Block) GetPrevRandao() common.Hash {
+	return b.prevRandao
+}
+
+// ComputePrevRandao computes the prevRandao from event hashes.
+func ComputePrevRandao(events []hash.Event) common.Hash {
+	prevRandao := common.Hash{}
+	for _, event := range events {
+		for i := 0; i < 24; i++ {
+			// first 8 bytes should be ignored as they are not pseudo-random.
+			prevRandao[i+8] = prevRandao[i+8] ^ event[i+8]
+		}
+	}
+	return sha256.Sum256(prevRandao.Bytes())
```

Comment:
By the specification written up in docs, that also serve as the documentation for this feature, only the last 24 bytes are hashed. Here you are hashing all 32 bytes, including 8 leading 0s. I think we should fix this, because it  does not make a lot of sense to include some constant values in the hash.